framework/doc/content/source/ics/IntegralPreservingFunctionIC.md

```diff
@@ -0,0 +1,42 @@
+# IntegralPreservingFunctionIC
+
+!syntax description /ICs/IntegralPreservingFunctionIC
+
+## Description
+
+Sets an initial condition while preserving an integral.
+The [!param](/ICs/IntegralPreservingFunctionIC/function) is used to specify the overall "form" of the initial condition,
+such that the applied initial condition is
+
+\begin{equation}
+\label{eq:ic1}
+u=q_0f
+\end{equation}
+
+where $u$ is the variable, $f$ is the function, and $q_0$ is a scaling
+factor used to preserve a total [!param](/ICs/IntegralPreservingFunctionIC/magnitude) upon volume integration:
+
+\begin{equation}
+\label{eq:ic2}
+q_0=\frac{Q}{\int_\Omega fd\Omega}
+\end{equation}
+
+where $Q$ is the total magnitude and $\Omega$ is the domain of integration.
+
+## Example Input Syntax
+
+As an example, below we set a sinusoidal heat source with generic form
+$\sin{\left(\frac{\pi z}{H}\right)}$ for a total magnitude of 550 (upon volume
```

Comment:
I m asking about magnitude because of this. It fits nicely here, does it for a random other function